tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -1434,15 +1435,16 @@ StorageManager::load_array_schema_from_uri(
 
   // Deserialize
   ConstBuffer cbuff(&buffer);
-  auto deserialized_schema{ArraySchema::deserialize(&cbuff, schema_uri)};
+  shared_ptr<ArraySchema> array_schema;
 
-  // #TODO Update catch statement after later StatusException changes
   try {
-    return {Status::Ok(),
-            make_shared<ArraySchema>(HERE(), deserialized_schema)};
+    auto deserialized_schema{ArraySchema::deserialize(&cbuff, schema_uri)};
+    array_schema = make_shared<ArraySchema>(HERE(), deserialized_schema);
   } catch (const StatusException& e) {
     return {Status_StorageManagerError(e.what()), nullopt};
   }
+
+  return {Status::Ok(), array_schema};
```

Comment:
Optional: Move the return statement below the try/catch block inside the try block. That way you won't need to pre-define `array_schema`.